compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -260,11 +260,14 @@ prettyErr lfVersion err = case err of
     SSC.ExceptionError e -> DA.Pretty.string $ show e
 
 
+activeContracts :: SS.ScenarioResult -> S.Set TL.Text
+activeContracts result =
+    S.fromList (V.toList (SS.scenarioResultActiveContracts result))
+
 prettyResult :: SS.ScenarioResult -> DA.Pretty.Doc Pretty.SyntaxClass
 prettyResult result =
     let nTx = length (SS.scenarioResultScenarioSteps result)
-        activeContracts = S.fromList (V.toList (SS.scenarioResultActiveContracts result))
-        nActive = length $ filter (SS.isActive activeContracts) (V.toList (SS.scenarioResultNodes result))
+        nActive = length $ filter (SS.isActive (activeContracts result)) (V.toList (SS.scenarioResultNodes result))
```

Comment:
> I wonder if we should factor out `activeContracts : ScenarioResult -> S.Set TL.Text` and similarly for scenario error.

@moritzkiefer-da maybe I'm myopic, but with "scenario error", what are you referring to?

Reply:
I was thinking of this part https://github.com/digital-asset/daml/blob/6a1e3afb71601b1021eec3b5ae9bc3fc02cf5a75/compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs#L1097 so `activeContracts : ScenarioError -> S.Set TL.Text`